File: pset1/cash/cash.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>

float obter_real_positivo(void);

int main(void)
{
    float reais = obter_real_positivo();
    int centavos = round(reais * 100);
    int quantidade_moedas = 0;
    while (centavos > 0)
    {
        if (centavos >= 25)
        {
            quantidade_moedas++;
            centavos -= 25;
        }
        else if (centavos >= 10)
        {
            quantidade_moedas++;
            centavos -= 10;
        }
        else if (centavos >= 5)
        {
            quantidade_moedas++;
            centavos -= 5;
        }
        else
        {
            quantidade_moedas++;
            centavos--;
        }
    }

    printf("%i\n", quantidade_moedas);
}

float obter_real_positivo(void)
{
    float n;
    do
    {
        n = get_float("Insira o valor a ser trocado: ");
    }
    while (n <= 0);
    return n;
}

File: pset6/cash/cash.py
from cs50 import get_float


def main():
    cash = get_positive_float("Insira o valor a ser trocado: ")

    cents = round(cash * 100)

    coin_counter = 0

    while cents > 0:
        if cents >= 25:
            coin_counter += 1
            cents -= 25
        elif cents >= 10:
            coin_counter += 1
            cents -= 10
        elif cents >= 5:
            coin_counter += 1
            cents -= 5
        else:
            coin_counter += 1
            cents -= 1

    print(coin_counter)


def get_positive_float(msg):
    while True:
        value = get_float(msg)
        if value > 0:
            break

    return value


main()
